LCM/dsc/engine/dsc_host/dsc_host.c

```diff
@@ -0,0 +1,324 @@
+        return MI_RESULT_FAILED;
+    else
+        return MI_RESULT_OK;
+}
+
+void PrintVersion()
+{
+    MI_Char versionInfo[DSCHOST_VERSION_BUFF_SIZE];
+    GetDSCHostVersion(versionInfo, DSCHOST_VERSION_BUFF_SIZE);
+    Tprintf(versionInfo);
+}
+
+void PrintHelp()
+{
+    Tprintf(MI_T("Usage:\n"));
+    Tprintf(MI_T("dsc_host [--help] [--version]\n"));
+    Tprintf(MI_T("dsc_host <Output Folder Path> <Operation> [Operation Arguments] \n"));
+    Tprintf(MI_T("\n"));
+    Tprintf(MI_T("Supported Operation values are:\n"));
+    Tprintf(MI_T("  SendConfiguration <MOF Document Path>\n"));
+    Tprintf(MI_T("  SendConfigurationApply <MOF Document Path>\n"));
+    Tprintf(MI_T("  GetConfiguration [Configuration Document Path]\n"));
+    Tprintf(MI_T("  ApplyConfiguration\n"));
+    Tprintf(MI_T("  SendMetaConfigurationApply <MOF Document Path> [force]\n"));
+    Tprintf(MI_T("  GetMetaConfiguration\n"));
+    Tprintf(MI_T("  RollBack\n"));
+    Tprintf(MI_T("  TestConfiguration\n"));
+    Tprintf(MI_T("  PerformRequiredConfigurationChecks [flags value]\n"));
+    Tprintf(MI_T("  StopConfiguration [force]\n"));
+    Tprintf(MI_T("  PerformInventory\n"));
+    Tprintf(MI_T("  PerformInventoryOOB [MOF Document Path]\n"));
+    Tprintf(MI_T("\n"));
+    Tprintf(MI_T("Examples:\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendConfiguration ./new_config.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendConfigurationApply ./new_config.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output GetConfiguration ./GetAuditPolicy.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output ApplyConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendMetaConfigurationApply ./new_metaconfig.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output GetMetaConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output RollBack\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output TestConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformRequiredConfigurationChecks 1\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output StopConfiguration force\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformInventory\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformInventoryOOB ./Inventory.mof\n"));
+    Tprintf(MI_T("\n"));
+}
+
+int main(int argc, char *argv[])
+{
+    MI_Instance *extended_error = NULL;
+    MI_Result result = MI_RESULT_OK;
+    DscSupportedOperation current_operation = DscSupportedOperation_NOP;
+    JSON_Value *operation_result_root_value = NULL;
+    JSON_Value *operation_error_root_value = NULL;
+    char* operation_name;
+
+    // Check the user that has invoked the operation: root for DIY and omsagent for OMS
+#if defined(BUILD_OMS)
+    if (RunningAsRoot() == MI_TRUE)
```

Comment:
>RunningAsRoot [](start = 8, length = 13)

root user is not allowed to run dsc operation in omsconfig. only omsagent can run dsc operation. how are we checking that?

Reply:
we only check that we are not running as root in OMS (existing behavior).

---
In reply to: [263580433](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/541#discussion_r263580433) [](ancestors = 263580433)